packages/@n8n/task-runner/src/task-runner-sentry.ts

```diff
@@ -0,0 +1,63 @@
+import { Service } from '@n8n/di';
+import type { ErrorEvent, Exception } from '@sentry/types';
+import { ErrorReporter } from 'n8n-core';
+
+import { SentryConfig } from './config/sentry-config';
+
+/**
+ * Sentry service for the task runner.
+ */
+@Service()
+export class TaskRunnerSentry {
+	constructor(
+		private readonly config: SentryConfig,
+		private readonly errorReporter: ErrorReporter,
+	) {}
+
+	async initIfEnabled() {
+		if (!this.config.dsn) return;
+
+		await this.errorReporter.init({
+			serverType: 'task_runner',
+			dsn: this.config.dsn,
+			release: this.config.n8nVersion,
+			environment: this.config.environment,
+			serverName: this.config.deploymentName,
+			beforeSendFilter: this.filterOutUserCodeErrors,
+		});
+	}
+
+	async shutdown() {
+		if (!this.config.dsn) return;
+
+		await this.errorReporter.shutdown();
+	}
+
+	/**
+	 * Filter out errors originating from user provided code.
+	 * It is possible for users to create code that causes unhandledrejections
+	 * that end up in the sentry error reporting.
+	 */
+	filterOutUserCodeErrors = (event: ErrorEvent) => {
+		const error = event?.exception?.values?.[0];
+		if (!error) return false;
```

Comment:
I think we can remove this line as `isUserCodeError` will return `false` if `error` is nullish.